extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -467,7 +465,30 @@ public function getCuratedContentSection() {
 			throw new NotFoundApiException( 'No members' );
 		}
 
-		$this->response->setVal( 'items', $data[ 'items' ] );
+		$this->response->setVal( 'items', $data['items'] );
+	}
+
+	public function getMainPageDetailsAndAdsContext() {
+		$mainPageTitle = Title::newMainPage();
+		$mainPageArticleID = $mainPageTitle->getArticleID();
+		$article = Article::newFromID( $mainPageArticleID );
+
+		$articleDetails = $this->getArticleDetails( $article );
+
+		if ( empty( $articleDetails ) ) {
+			throw new WikiaException( 'Article details empty', 500 );
+		}
+
+		$adsContext = $this->mercuryApi->getAdsContext( $mainPageTitle );
+
+		if ( empty( $adsContext ) ) {
```

Comment:
so if I decide not to throw I would need to check return val of this function everywhere I call it in front end and reject the promise if data is empty instead of relying on automatic rejection when error is thrown?